hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java

```diff
@@ -244,7 +247,8 @@ && equal( join.getKey(), manyToOne ) ) {
 							final Value value = property.getValue();
 							if ( value instanceof OneToOne ) {
 								final OneToOne oneToOne = (OneToOne) value;
-								if ( name.equals( oneToOne.getMappedByProperty() )
+								if (declaringTableGroupProducer.getNavigableRole().getLocalName().equals( oneToOne.getReferencedEntityName() )
+								&& name.equals( oneToOne.getMappedByProperty() )
```

Comment:
I have the feeling that this will miss out on properties that are part of embeddables i.e. we will never detect the `bidirectionalAttributeName` for a `@OneToOne` that uses e.g. `mappedBy = "embeddable.abc"` because `name` will never contain the `.`. IMO this should rather use the local name of the navigable role to determine the bidirectional "path". That thing possibly being a path is the second thing I'm concerned with. Our current uses of `bidirectionalAttributeName` do not take into account that this could be a path i.e. we should use `findByPath` instead of `findSubPart`. Also, some of the current `isBidirectionalAttributeName()` logic assumes that it can compare `bidirectionalAttributeName` against a `NavigablePath#getLocalName()`, which isn't the case if we admit that it is a path.
I suggest we rather model this as `SelectablePath bidirecationalAttributePath` and adapt implementations accordingly.